p2p/gossip/protocol.go

```diff
@@ -216,58 +252,74 @@ func (s *Neighborhood) getMorePeers(numpeers int) {
 	nds := s.ps.SelectPeers(numpeers)
 	ndsLen := len(nds)
 	if ndsLen == 0 {
-		go func() {
-			time.Sleep(1 * time.Second)
-			s.morePeersReq <- struct{}{}
-		}()
+		s.Debug("Peer sampler returned nothing.")
 		// this gets busy at start so we spare a second
-		return // we cant connect if we don't have peers
+		time.Sleep(NoResultsTimeout)
```

Comment:
this logic should be handled by the caller of this method. The caller should decide whether or not it wishes to sleep and retry